src/main/java/com/govac/institutii/kafka/Producer.java

```diff
@@ -0,0 +1,57 @@
+package com.govac.institutii.kafka;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+@Component
+public class Producer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Producer.class);
+
+    private final KafkaProducer<byte[], byte[]> producer;
+
+    public Producer(Properties kafkaProperties) throws IOException {
+        producer = new KafkaProducer<byte[], byte[]>(kafkaProperties);
+    }
+
+    public void send(Message message) {
+        producer.send(new ProducerRecord<byte[], byte[]>(message.getTopic(), message.getKey(), message.getValue()),
+                callBack(message));
```

Comment:
metoda send() din clientul Kafka returneaza un Future pe care se poate apela imediat metoda get(), asta face ca threadul sa se blocheze pana primeste raspuns (sincron). Metoda de send de aici nu propaga Future-ul e drept - o sa schimb semnatura ca sa il propag.
Abordarea asincrona mi se pare mai flexibila, pentru ca ne da voie sa avem o rata de transfer mult mai buna (in cazul multor mesaje care trebuie trimise simultan) si ne da voie sa fim toleranti la erori (in cazul in care nu se pot trimite mesajele putem sa reincercam trimiterea lor sau sa le salvam pentru procesarea ulterioara).